code/datums/abilities/ability_parent.dm

```diff
@@ -1191,6 +1197,12 @@
 		for (var/datum/abilityHolder/H in holders)
 			H.deductPoints(cost)
 
+	addPoints(add_points, var/datum/abilityHolder/w_list)
+		for (var/datum/abilityHolder/H in holders)
+			if (w_list && !istype(H, w_list))
+				continue
+			H.addPoints(add_points)
```

Comment:
```suggestion
	addPoints(add_points, target_ah_type)
		for (var/datum/abilityHolder/H in holders)
			if (w_list && !istype(H, target_ah_type))
				continue
			H.addPoints(add_points)
```
`w_list` is not a very descriptive name, moreover you don't actually want an abilityHolder *object* to get passed here but a *type* so the argument should be untyped.